### ISSUE TITLE
Unable to build with Python 3.7.6

### ISSUE THREAD

**Issue by feerrenrut** [nvaccess/nvda#10696]
### Steps to reproduce:
Install Python 3.7.6 and try to rebuild NVDA (from clean working directory)

### Actual behavior:
We have had several reports of errors ending with: `TypeError: item 2 in _argtypes_ passes a union by value, which is unsupported.`

### Expected behavior:
Build should complete successfully.

#### NVDA installed/portable/running from source:
Source

#### Name and version of other software in use when reproducing the issue:
Python 3.7.6
Reverting to 3.7.5 seems to result in a successful build.

#### Error:

```
c:\nvda>scons source
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
interfaceAction(["source\comInterfaces\AcrobatAccessLib.py", "source\comInterfaces\__pycache__\AcrobatAccessLib.cpython-37.pyc"], ["source\typelibs\AcrobatAccess.tlb"])
scons: *** [source\comInterfaces\AcrobatAccessLib.py] TypeError : item 2 in _argtypes_ passes a union by value, which is unsupported.
Traceback (most recent call last):
  File "C:\Python37-32\lib\ctypes\__init__.py", line 121, in WINFUNCTYPE
    return _win_functype_cache[(
restype, argtypes, flags)]
KeyError: (<class 'ctypes.HRESULT'>, (<class 'ctypes.c_long'>, <class 'comtypes.automation.tagVARIANT'>), 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\nvda\include\scons\src\engine\SCons\Action.py", line 1202, in execute
    result = self.execfunction(target=target, source=rsources, env=env)
  File "C:\nvda\source\comInterfaces_sconscript", line 35, in interfaceAction
    comtypes.client.GetModule(source)
  File "c:\nvda\include\comtypes\comtypes\client\_generate.py", line 110, in GetModule
    mod = _CreateWrapper(tlib, pathname)
  File "c:\nvda\include\comtypes\comtypes\client\_generate.py", line 184, in _CreateWrapper
    mod = _my_import(fullname)
  File "C:\nvda\source\comInterfaces_sconscript", line 26, in new_my_import
    return old_my_import(fullname)
  File "c:\nvda\include\comtypes\comtypes\client\_generate.py", line 24, in _my_import
    return __import__(fullname, globals(), locals(), ['DUMMY'])
  File "C:\nvda\source\comInterfaces\_C523F390_9C83_11D3_9094_00104BD0D535_0_3_0.py", line 353, in <module>
    ( ['in'], VARIANT, 'value' )),
  File "c:\nvda\include\comtypes\comtypes\__init__.py", line 329, in __setattr__
    self._make_methods(value)
  File "c:\nvda\include\comtypes\comtypes\__init__.py", line 698, in _make_methods
    prototype = WINFUNCTYPE(restype, *argtypes)
  File "C:\Python37-32\lib\ctypes\__init__.py", line 123, in WINFUNCTYPE
    class WinFunctionType(_CFuncPtr):
TypeError: item 2 in _argtypes_ passes a union by value, which is unsupported.
scons: building terminated because of errors.
```

**Comment by josephsl** [comment]
Hi, I can build the source to completion in 3.7.6 on my computer. Thanks.

**Comment by josephsl** [comment]
Never mind – it does fail on my system as well. Thanks.

**Comment by DrSooom** [comment]
@feerrenrut: Maybe two typos in the issue description: Python 2.7.x instead of Python 3.7.x.

**Comment by feerrenrut** [comment]
Thanks @DrSooom I have amended it.

**Comment by josephsl** [comment]
Hi,

3.7.7 should fix this issue, see:

https://github.com/python/cpython/pull/17960

Thanks.

**Comment by josephsl** [comment]
Hi,

According to a Python committer who reported on Python issue 16575 (not GitHub), comtypes is the "real" culprit for us. The overall issue has to do with libffi which is the basis for ctypes.

Thanks.

**Comment by feerrenrut** [comment]
Thanks for investigation @josephsl

**Comment by lukaszgo1** [comment]
While developers can downgrade to 3.7.5 easily enough  do we have any plan for the time when 3.7.6  would be deployed on AppVeyor?

**Comment by feerrenrut** [comment]
If it becomes necessary, we can specify the python version via the `appveyor.yml` file. We currently have `environment:  PY_PYTHON: 3.7-32`
See also: https://help.appveyor.com/discussions/problems/10557-unexpected-python-version

**Comment by josephsl** [comment]
Hi,

Resolved in Python 3.7.7 (released on March 10, 2020). Closing.

Thanks.
